lib/babe/epoch.go

```diff
@@ -41,24 +41,18 @@ func (b *Service) initiateEpoch(epoch uint64) error {
 			return err
 		}
 
-		var data *types.EpochData
 		if !has {
-			data = &types.EpochData{
-				Randomness:  b.epochData.randomness,
-				Authorities: b.epochData.authorities,
-			}
-
-			err = b.epochState.SetEpochData(epoch, data)
-		} else {
-			data, err = b.epochState.GetEpochData(epoch)
+			logger.Crit("no epoch data for next BABE epoch", "epoch", epoch)
+			return errNoEpochData
 		}
 
+		data, err := b.epochState.GetEpochData(epoch)
 		if err != nil {
 			return err
 		}
 
 		idx, err := b.getAuthorityIndex(data.Authorities)
-		if err != nil && err != ErrNotAuthority {
+		if err != nil && err != ErrNotAuthority { // TODO: this should be checked in the upper function
```

Comment:
Use `errors.Is` to check for a specific error

Reply:
done